### PR TITLE
Try upgrade cadl compiler lib version

### LINKED ISSUE #1498: use latest cadl compiler 0.34.0 and cadl-rest 0.16.0

### PR DESCRIPTION
**Pull request by qiaozha**
fix #1498

### CODE REVIEW

File: packages/cadl-rlc-emitter/src/modelUtils.ts [hunk 1]
```diff
@@ -688,13 +689,13 @@ function mapCadlIntrinsicModelToTypeScript(
...
     case "duration":
       return { type: "string", format: "duration" };
-    case "Map":
-      // We assert on valType because Map types always have a type
-      const valType = (cadlType as ModelType)?.properties.get("v");
-      return {
-        type: "object",
-        additionalProperties: getSchemaForType(program, valType!.type, usage)
-      };
+    // case "Map":
```
Comment (joheredi) [inline packages/cadl-rlc-emitter/src/modelUtils.ts:692]: I think Map changed Record

File: packages/cadl-rlc-emitter/src/transform/transformParameters.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ export function transformToParameterTypes(
   for (const route of routes) {
     const parameters = route.parameters;
     const rlcParameter: OperationParameter = {
-      operationGroup: route.groupName,
+      operationGroup: route.operation.namespace?.name ?? "",
```
Comment (joheredi) [inline packages/cadl-rlc-emitter/src/transform/transformParameters.ts:34]: I wonder if we need to handle or error out if there is a name missing?

File: packages/cadl-rlc-emitter/src/transform/transformSchemas.ts [hunk 3]
```diff
@@ -11,15 +11,15 @@ export function transformSchemas(program: Program) {
...
+    if (route.parameters.bodyParameter) {
+      const bodyModel = route.parameters.bodyType;
+      if (bodyModel && bodyModel.kind === "Model") {
         setModelMap(bodyModel, SchemaContext.Input);
-      } else if (
-        bodyModel.kind === "Array" &&
-        bodyModel.elementType.kind === "Model"
-      ) {
-        setModelMap(bodyModel.elementType, SchemaContext.Input);
+      // } else if (
```
Comment (joheredi) [inline packages/cadl-rlc-emitter/src/transform/transformSchemas.ts:18]: Are we no longer handling Array?

File: packages/cadl-rlc-emitter/src/modelUtils.ts [hunk 4]
```diff
@@ -614,6 +592,29 @@ function mapCadlIntrinsicModelToTypeScript(
...
+          type: "array",
+          items: getSchemaForType(program, indexer.value!),
+        };
+      }
+      if (usage && usage.includes(SchemaContext.Output)) {
+        schema.outputTypeName = `Array<${schema.items.name}Output>`;
+      }
+      schema.usage = usage;
+      return schema;
+    }
```
Comment (qiaozha) [inline packages/cadl-rlc-emitter/src/modelUtils.ts:616]: array and map is being handled here.

File: packages/cadl-rlc-emitter/src/transform/transformParameters.ts (hunk 2 above)
Comment (qiaozha) [reply packages/cadl-rlc-emitter/src/transform/transformParameters.ts:34]: not sure if cadl has client level operations.

File: packages/cadl-rlc-emitter/src/modelUtils.ts (hunk 1 above)
Comment (qiaozha) [reply packages/cadl-rlc-emitter/src/modelUtils.ts:692]: Okay, let me double check.

**Review by joheredi — APPROVED** [review]

File: packages/cadl-rlc-emitter/src/modelUtils.ts (hunk 1 above)
Comment (qiaozha) [reply packages/cadl-rlc-emitter/src/modelUtils.ts:692]: @joheredi I can't find record in the IntrinsicModelName definition here https://github.com/microsoft/cadl/blob/main/packages/compiler/core/types.ts#L178-L200.

File: packages/cadl-rlc-emitter/src/transform/transformParameters.ts (hunk 2 above)
Comment (MaryGao) [reply packages/cadl-rlc-emitter/src/transform/transformParameters.ts:34]: Hi @mikekistler I'd like to confirm with you below two questions

- Is it possible that a cadl file without `namespace` ?
- Will an operation not belong to any `namespace` ? I know in swag spec we could have client-level operations without operation group as its prefix.

Comment (mikekistler) [reply packages/cadl-rlc-emitter/src/transform/transformParameters.ts:34]: @MaryGao Cadl namespaces are intended for designating operation groups. A new set of client decorators has been developed for that purpose. See:

https://github.com/Azure/cadl-azure/blob/main/packages/cadl-azure-core/README.md#define-clients-and-subclients

**Review by joheredi — APPROVED** [review]

**Pull request merged**